src/include/intarith.h

```diff
@@ -57,6 +57,18 @@ constexpr inline T p2align(T x, T align) {
   return x & -align;
 }
 
+/*
+ * return whether x is aligned with (1 << bits)
+ * eg, p2aligned(1200, 10) ==> false
+ * eg, p2align(1024, 10) ==> true
+ * eg, p2align(0x1234, 8) ==> false
+ * eg, p2align(0x5600, 8) ==> true
+ */
+template<typename T>
+constexpr inline bool p2_isaligned(T x, unsigned bits) {
```

Comment:
1) naming standard is p2isaligned
2) every function in p2* has second parameter that is direct value; this new function uses bits; thats error prone